src/network/dns/resolver.rs

```diff
@@ -11,8 +14,26 @@ pub trait Lookup {
 pub struct Resolver(TokioAsyncResolver);
 
 impl Resolver {
-    pub async fn new(runtime: Handle) -> Result<Self, failure::Error> {
-        let resolver = TokioAsyncResolver::from_system_conf(runtime).await?;
+    pub async fn new(
+        runtime: Handle,
+        dns_server: &Option<Ipv4Addr>,
+    ) -> Result<Self, failure::Error> {
+        let resolver = match dns_server {
+            Some(dns_server_address) => {
+                let mut config = ResolverConfig::new();
+                let options = ResolverOpts::default();
+                // let socket = SocketAddr::V4(SocketAddrV4::new(Ipv4Addr::new(1, 1, 1, 1), 53));
+                let socket = SocketAddr::V4(SocketAddrV4::new(*dns_server_address, 53));
+                let nameserver_config = NameServerConfig {
+                    socket_addr: socket,
+                    protocol: Protocol::Udp,
+                    tls_dns_name: None,
+                };
+                config.add_name_server(nameserver_config);
+                TokioAsyncResolver::new(config, options, runtime).await?
```

Comment:
It's a shame we can't just change the nameserver of the configuration returned by `from_system_conf`, but I think this is the best we can do right now :)

Reply:
Yeah, for sure. I mostly want this feature as a workaround for issues where we fail to parse resolv.conf, so we'd be panicking by then.